capa/render/proto/__init__.py

```diff
@@ -0,0 +1,445 @@
+# Copyright (C) 2023 FireEye, Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at: [package root]/LICENSE.txt
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import io
+import sys
+from typing import Dict, Union
+from dataclasses import dataclass
+
+import pydantic
+
+import capa.render
+import capa.render.utils
+import capa.features.freeze
+import capa.render.result_document
+import capa.features.freeze.features
+from capa.render.utils import StringIO
+
+
+def emit_proto_enum(out: StringIO, enum):
+    # like: AddressType
+    title = enum["title"]
+
+    # like: ADDRESSTYPE
+    prefix = title.upper()
+
+    def render_value(value):
+        # like: ADDRESSTYPE_ABSOLUTE
+        return "%s_%s" % (prefix, value.upper().replace(" ", "_"))
+
+    # like:
+    #
+    #     enum AddressType {
+    #         ADDRESSTYPE_UNSPECIFIED = 0;
```

Comment:
in initial testing these enum values vs. our address type strings result in an error:
```
google.protobuf.json_format.ParseError: Failed to parse meta field: Failed to parse analysis field: Failed to parse base_address field: Failed to parse type field: Invalid enum value absolute for enum type AddressType at ResultDocument.meta.analysis.ba
se_address.type....
```

Reply:
yeah, i don't think this will work with the built-in json translation because the proto format doesnt support exactly our json structure. i'm working on a translator right now.